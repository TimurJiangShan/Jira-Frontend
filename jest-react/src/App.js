import React from 'react';

function App() {
  return (
    <div className="app-container" title="dell lee" data-test='container'>
      Hello World !
    </div>
  );
}

export default App;
