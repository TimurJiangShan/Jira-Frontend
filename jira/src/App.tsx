import { AuthenticatedApp } from "AuthenticatedApp";
import React from "react";
import "./App.css";
import { UnauthenticatedAPP } from "unauthenticatedApp";
import { useAuth } from "./context/auth-context";

function App() {
  const { user } = useAuth();
  return (
    <div className="App">
      {user ? <AuthenticatedApp /> : <UnauthenticatedAPP />}
    </div>
  );
}

export default App;
