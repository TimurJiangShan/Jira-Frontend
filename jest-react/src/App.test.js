import React from 'react';
import App from './App';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
// mount: 当前组件以及子组件全渲染出来
// 单元测试用 shallow, 集成测试用 mount

Enzyme.configure({ adapter: new Adapter() });

it('renders without crashing', () => {
  const wrapper = mount(<App />);
  expect(wrapper).toMatchSnapshot();
  // const container = wrapper.find('[data-test="container"]');
  // expect(wrapper.find('[data-test="container"]').length).toBe(1);
  // console.log(wrapper.debug());
  // expect(wrapper.find('[data-test="container"]').prop('title')).toBe('dell lee');
  // expect(container).toExist();
  // expect(container).toHaveProp('title', 'dell lee');
});
