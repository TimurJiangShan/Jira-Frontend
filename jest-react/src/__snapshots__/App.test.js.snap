// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<App>
  <div
    className="app-container"
    data-test="container"
    title="dell lee"
  >
    Hello World !
  </div>
</App>
`;
